src/daemon/common.c

```diff
@@ -262,10 +262,7 @@ int swrite(int fd, const void *buf, size_t count) {
   pfd.events = POLLIN | POLLHUP;
   pfd.revents = 0;
   if (poll(&pfd, 1, 0) > 0) {
-    char buffer[32];
-    if (recv(fd, buffer, sizeof(buffer), MSG_PEEK | MSG_DONTWAIT) == 0) {
-      /* if recv returns zero (even though poll() said there is data to be
-       * read), that means the connection has been closed */
+    if (pfd.revents & POLLHUP) {
```

Comment:
Unfortunately, my check shows this change not equivalent to existing implementation.

Reply:
I'm having trouble recreating this issue. Running the write_graphite plugin, I actually had the opposite problem where the pfd.revents & POLLHUP block is executing and not the recv(...) == 0 block, but I think I have a fix for that.

With the memcached plugin enabled for me, neither seems to be called.

What is your configuration?

Reply:
>Running the write_graphite plugin, I actually had the opposite problem where the pfd.revents & POLLHUP block is executing and not the recv(...) == 0 block, but I think I have a fix for that.

That is caused by a fact what `write_graphite` writes mostly, and `memcached` mostly reads.
As I say before,
>It [POLLHUP] appears there only after write() attempt, so I see no way to code changes.
That may be possible what for `write_graphite`, you found `POLLHUP` set after previous write attempt has failed.

```
ReadThreads  1
WriteThreads 1
MaxReadInterval 20

LoadPlugin memcached
<Plugin memcached>
       <Instance "local">
               Address "127.0.0.1"
               Port "11211"
               #Socket "/var/run/memcached/memcached.sock"
       </Instance>
</Plugin>

+network plugin as write plugin
+logfile plugin to get console output 
+configure --enable-debug of course
```

I used a memcached instance or 'nc -l 127.0.0.1 -p 11211' for debug.
Also I use a `strace` for debugging: 
Look for main Collectd pid, then do 'cd /proc/PID/tasks', 'ls -l'. 
Last one (with max value in my case) will be a PID of read thread.